File: project_euler/03_lar_pri_num.rb
#!/usr/bin/ruby -w

require 'prime'

big_num = 600851475143
big_num_sqrt = Math.sqrt(big_num).floor

while big_num_sqrt > 1
  if Prime.prime?(big_num_sqrt) && big_num % big_num_sqrt == 0
    if Prime.prime?(big_num / big_num_sqrt) 
      puts big_num / big_num_sqrt
    else
      puts big_num_sqrt
    end
    break
  end
  big_num_sqrt -= 1
end

File: project_euler/03_lar_pri_num.py
'''
The prime factors of 13195 are 5, 7, 13 and 29.

What is the largest prime factor of the number 600851475143 ?
'''
import math
import PrimeValidate

bigNum = 600851475143
bigNumSqrt = int(math.floor(math.sqrt(bigNum))) - 1
hiEnd = bigNumSqrt
while (hiEnd > 2):
  if (bigNum % hiEnd == 0) and PrimeValidate.isPrime(hiEnd): 
    print hiEnd
    break
  else:
    hiEnd -= 1
